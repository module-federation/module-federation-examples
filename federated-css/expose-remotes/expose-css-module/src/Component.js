import React from 'react';
import classes from './Button.styles.module.css';

const style = {
  padding: 12,
  backgroundColor: '#cccccc',
};

const Component = () => (
  <div style={style}>
    <section className="buttons">
      <div className={classes.redButton} data-e2e="FEDERATED_CSS_BUTTON">
        Css Module styled Button
      </div>
    </section>
  </div>
);

export default Component;
