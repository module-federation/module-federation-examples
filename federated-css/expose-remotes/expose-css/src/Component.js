import React from 'react';
import './Button.css';

const style = {
  padding: 12,
  backgroundColor: '#cccccc',
};

const CssButtonContainer = () => (
  <div style={style}>
    <section className="buttons">
      <div className="red-button" data-e2e="FEDERATED_CSS_BUTTON">
        Css styled Button
      </div>
    </section>
  </div>
);

export default CssButtonContainer;
