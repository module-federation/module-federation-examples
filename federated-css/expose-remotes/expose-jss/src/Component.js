import React from 'react';
import styles from './react-jss.styles';

const Button = () => {
  const classes = styles();

  return (
    <div className={classes.button} data-e2e="FEDERATED_CSS_BUTTON">
      Federated Button with Jss styling
    </div>
  );
};

export default Button;
