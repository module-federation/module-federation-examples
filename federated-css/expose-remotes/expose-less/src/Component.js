import React from 'react';
import './Button.styles.less';

const style = {
  padding: 12,
  backgroundColor: '#cccccc',
};

const Component = () => (
  <div style={style}>
    <section className="buttons">
      <div className="brown-button" data-e2e="FEDERATED_CSS_BUTTON">
        Less styled Button
      </div>
    </section>
  </div>
);

export default Component;
