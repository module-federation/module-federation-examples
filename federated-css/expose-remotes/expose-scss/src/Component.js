import React from 'react';
import './Button.styles.scss';

const style = {
  padding: 12,
  backgroundColor: '#cccccc',
};

const Component = () => (
  <div style={style}>
    <section className="buttons">
      <div className="orange-button" data-e2e="FEDERATED_CSS_BUTTON">
        Scss styled Button
      </div>
    </section>
  </div>
);

export default Component;
