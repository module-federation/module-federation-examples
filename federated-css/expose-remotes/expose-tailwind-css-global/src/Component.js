import React from 'react';
import './tailwind.global.css';

const style = {
  padding: 12,
  backgroundColor: '#cccccc',
};

const Component = () => (
  <div style={style}>
    <section className="buttons">
      <div className="btn btn-green" data-e2e="FEDERATED_CSS_BUTTON">
        Tailwind Global import styled Button
      </div>
    </section>
  </div>
);

export default Component;
