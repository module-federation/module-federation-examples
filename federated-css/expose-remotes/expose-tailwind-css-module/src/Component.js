import React from 'react';
import classes from './tailwind.module.css';

const style = {
  padding: 12,
  backgroundColor: '#cccccc',
};

const Component = () => (
  <div style={style}>
    <section className="buttons">
      <div className={`${classes.btn} ${classes.btnBlue}`} data-e2e="FEDERATED_CSS_BUTTON">
        Tailwind Css Module styled Button
      </div>
    </section>
  </div>
);

export default Component;
