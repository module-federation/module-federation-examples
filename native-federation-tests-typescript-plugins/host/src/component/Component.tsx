import { AnotherButton } from 'moduleFederationTypescript/anotherButton'
import Button from 'moduleFederationTypescript/button'

const Component = () => <div>
    <Button onClick={console.log} />
    <AnotherButton onClick={console.error} />
</div>

export default Component