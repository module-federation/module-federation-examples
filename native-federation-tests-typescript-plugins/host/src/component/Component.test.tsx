import { render } from '@testing-library/react'
import { describe, expect, it, vi } from 'vitest'
import Component from './Component'

vi.mock('moduleFederationTypescript/button')
vi.mock('moduleFederationTypescript/anotherButton')

describe('Component', () => {
    it('can test federated components', () => {
        const { asFragment } = render(<Component />)

        expect(asFragment()).toMatchSnapshot()
    })
})