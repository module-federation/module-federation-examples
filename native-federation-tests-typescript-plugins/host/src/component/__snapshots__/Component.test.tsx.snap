// Vitest Snapshot v1

exports[`Component > can test federated components 1`] = `
<DocumentFragment>
  <div>
    <button>
      Federated button
    </button>
    <button>
      Another federated button
    </button>
  </div>
</DocumentFragment>
`;
